syntax = "proto3";

message Cmd {
    uint32 id = 1;
    oneof content {
        PIDCmd pid_cmd = 2;
    }
}

message PIDCmd {
    // PID gains
    bool enable_pid = 1;
    float kp = 2;
    float ki = 3;
    float kd = 4;
    // Anti-windup
    bool enable_anti_windup = 5;
    float min = 6;
    float max = 7;
    // Sampling period [s]
    float ts = 8;
    // Setpoint
    float w = 9;
    // controller output
    float u = 10;
    // first order compensation time constant
    float tau = 11;
}
